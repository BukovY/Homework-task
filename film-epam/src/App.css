body{
  width: 990px;
  margin: 0 auto;
  background: #151515;
  color: white;
}
.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
.film_card_grid{
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-gap: 10px;
}
