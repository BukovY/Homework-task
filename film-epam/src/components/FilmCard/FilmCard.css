.card .title{
    font-size: 18px;
}
.card .genres{
    font-size: 14px;
}
.film_cover {
    max-height: 290px;
}
.film_cover img {
width: 100%;

}
.film_cover .play_film {
    visibility: hidden;
    width: 90px;
    position: relative;
    bottom: 270px;
    left: calc(50% - 45px);

}
.film_cover:hover .play_film {
    visibility: visible;
    width: 90px;
    bottom: 200px;
    left: calc(50% - 45px);
    cursor: pointer;
}

.rating {
    position: relative;
    top: 35px;
    left: -5px;
    width: 30px;
    text-align: center;
    border-radius: 5px;
}
.rating_up {
    background-color: green;
}
.rating_down {
    background-color: gray;
}