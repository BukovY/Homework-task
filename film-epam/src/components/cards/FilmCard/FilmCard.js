import React from "react";

function FilmCard({props}) {
    return (<div>
            <p>{props.title}</p>
        </div>
    );
}

export default FilmCard;