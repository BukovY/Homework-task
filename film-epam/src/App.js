import React from "react";
import './App.css';
import {store} from "./redux/store";
import Homepage from "./Pages/Homepage/Homepage";
import Header from "./components/Header/Header";

function App() {
    const page = store.selectPage
    return (
        <div>
            <Header/>
            {page === 'main' ? <Homepage/> : ''}
            {page === 'actor' ? '' : ''}
            {page === 'movie' ? '' : ''}

        </div>
    );
}

export default App;
