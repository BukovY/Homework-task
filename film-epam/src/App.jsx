import React, { useEffect } from "react";
import "./App.css";
import Header from "./components/Header/Header";
import { ErrorBoundary } from "./components/ErrorBoundary/Error Boundary";
import HomePage from "./pages/HomePage/HomePage.jsx";
import { useDispatch, useSelector } from "react-redux";
import MoviePage from "./pages/MoviePage/MoviePage.jsx";
import ActorPage from "./pages/ActorPage/ActorPage";
import SearchPage from "./pages/SearchPage/SearchPage";
import { getGenresMap, getFilmsData } from "./redux/reducers/appReducers";
import { getFilm } from "./redux/reducers/movieReducers";
import { getActorInfo } from "./redux/reducers/actorReducers";
import { getSearchData } from "./redux/reducers/searchReducers";
import LoaderPlaceholder from "./components/LoarerPlaceholder/LoaderPlaceholder";

const App = () => {
  const dispatch = useDispatch();

  const {
    activeFilter,
    languageSelected,
    paginationPage,
    page,
    search,
    isFetching,
  } = useSelector((state) => state.appReducer);

  const { selectedMovie, fetchingFilm } = useSelector(
    (state) => state.movieReducers
  );

  const { actorId, fetchingActor } = useSelector(
    (state) => state.actorReducers
  );

  const { searchPage, needUpdate, fetchingSearch } = useSelector(
    (state) => state.searchReducers
  );

  useEffect(() => {
    if (selectedMovie !== "") {
      const input = {
        selectedMovie: selectedMovie,
        languageSelected: languageSelected,
      };
      dispatch(getFilm(input));
    }
  }, [selectedMovie, languageSelected]);

  useEffect(() => {
    if (actorId !== "") {
      const inputs = {
        actorId: actorId,
        languageSelected: languageSelected,
      };
      dispatch(getActorInfo(inputs));
    }
  }, [languageSelected, actorId]);

  useEffect(() => {
    dispatch(getGenresMap(languageSelected));
  }, [languageSelected]);

  useEffect(() => {
    const inputs = {
      activeFilter: activeFilter,
      languageSelected: languageSelected,
      paginationPage: paginationPage,
    };
    dispatch(getFilmsData(inputs));
  }, [activeFilter, languageSelected, paginationPage]);

  useEffect(() => {
    if (needUpdate) {
      const input = {
        search,
        searchPage,
        languageSelected,
      };
      dispatch(getSearchData(input));
    }
  }, [searchPage, needUpdate, languageSelected]);

  return (
    <div>
      <Header />

      {isFetching || fetchingFilm || fetchingActor || fetchingSearch ? (
        <LoaderPlaceholder />
      ) : (
        ""
      )}
      {page === "search" && !fetchingSearch ? (
        <ErrorBoundary>
          <SearchPage />
        </ErrorBoundary>
      ) : (
        ""
      )}
      {page === "main" && !isFetching ? (
        <ErrorBoundary>
          <HomePage />
        </ErrorBoundary>
      ) : (
        ""
      )}

      {page === "movie" && !fetchingFilm ? (
        <ErrorBoundary>
          <MoviePage />
        </ErrorBoundary>
      ) : (
        ""
      )}

      {page === "actor" && !fetchingActor ? (
        <ErrorBoundary>
          <ActorPage />
        </ErrorBoundary>
      ) : (
        ""
      )}
    </div>
  );
};

export default App;
