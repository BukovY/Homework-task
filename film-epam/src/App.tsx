import { ErrorBoundary } from "./components/ErrorBoundary";
import "./App.css";
import { Header } from "./components/Header";
import HomePage from "./pages/HomePage/HomePage";
import MoviePage from "./pages/MoviePage/MoviePage";
import ActorPage from "./pages/ActorPage/ActorPage";
import SearchPage from "./pages/SearchPage/SearchPage";
import { Route, Switch } from "react-router";
import NotFoundPage from "./pages/NotFoundPage/NotFoundPage";
import React, { FC } from "react";

const App: FC = () => {
  return (
    <>
      <Header />
      <Switch>
        <Route path="/" exact>
          <ErrorBoundary>
            <HomePage />
          </ErrorBoundary>
        </Route>
        <Route path="/movie/:id">
          <ErrorBoundary>
            <MoviePage />
          </ErrorBoundary>
        </Route>
        <Route path="/actor/:id">
          <ErrorBoundary>
            <ActorPage />
          </ErrorBoundary>
        </Route>
        <Route path="/search/:id">
          <ErrorBoundary>
            <SearchPage />
          </ErrorBoundary>
        </Route>
        <Route>
          <NotFoundPage />
        </Route>
      </Switch>
    </>
  );
};

export default App;
