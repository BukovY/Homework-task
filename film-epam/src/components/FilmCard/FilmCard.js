import React from "react";
import './FilmCard.css'
import play from '../../src/img/play.svg'

function FilmCard({img, rating, title, genres}) {

    return (
        <div className='card'>
            <div className={rating < 7 ? 'rating rating_down' : 'rating rating_up'}>{rating}</div>
            <div className='film_cover'>
                <img src={`https://image.tmdb.org/t/p/w500/${img}`} className='film_cover' alt={title} />
                <img src={play} width='50px' className='play_film' alt='play' />
            </div>
            {title? <div className='title'>{title}</div> : ''}
            {genres? <div className='genres'>{genres.join(" ")}</div> : ''}
        </div>
    );
}

export default FilmCard;