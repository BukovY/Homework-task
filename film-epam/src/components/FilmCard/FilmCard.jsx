import React from "react";
import "./FilmCard.css";
import play from "../../static/img/play.svg";
import { store } from "../../redux/store";
import { rerender } from "../../index";

const FilmCard = ({ img, rating, title, genres, id }) => {
  const openFilm = (id) => {
    store.selectedFilm = id;
    store.isTooltipLanguageOpen = false;
    store.selectPage = "movie";
    rerender();
  };
  if(img == null){
      img = 'https://upload.wikimedia.org/wikipedia/commons/6/65/No-Image-Placeholder.svg'
  } else {
      img = `https://image.tmdb.org/t/p/w500/${img}`
  }
  return (
    <div className="card">
      <div className={rating < 7 ? "rating rating_down" : "rating rating_up"}>
        {rating}
      </div>
      <div className="film_cover">
        <img
          src={img}
          className="film_cover"
          alt={title}
          onClick={() => openFilm(id)}
        />
        <img src={play} width="50px" className="play_film" alt="play" />
      </div>
      {title ? <div className="title">{title}</div> : ""}
      {genres ? <div className="genres">{genres.join(" ")}</div> : ""}
    </div>
  );
};

export default FilmCard;
