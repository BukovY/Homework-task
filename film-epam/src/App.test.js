import { render, screen, fireEvent } from "@testing-library/react";
import React from "react";
import App from "./App";
import store from "./redux/store";
import { Provider } from "react-redux";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import { PhotoCard } from "./components/PhotoCard";
import { Search } from "./components/Search";
import { Paginations } from "./components/Pagination";
import { MetaBlock } from "./components/MetaBlock";
import { IncorrectRequest } from "./components/IncorrectRequest";
import { LanguageSelect } from "./components/Language/components/LanguageSelect";
import { Pagination } from "./components/Pagination/components/Parination";
import { People } from "./components/People";
import { FilmCard } from "./components/FilmCard";
import { FilmCover } from "./components/FilmCover";
import { Tab } from "./components/Tabs/components/Tab";

const wrapper = (component) => {
  return (
    <Provider store={store}>
      <BrowserRouter>{component}</BrowserRouter>
    </Provider>
  );
};

const routerWrapper = (path) => {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={[path]}>
        <App />
      </MemoryRouter>
    </Provider>
  );
};
describe("Markup testing", () => {
  it("Tab in document ", () => {
    render(wrapper(<App />));
    screen.debug();
    expect(screen.getByText(/Popular/i)).toBeInTheDocument();
    expect(screen.getByText(/Top rated/i)).toBeInTheDocument();
    expect(screen.getByText(/Upcoming/i)).toBeInTheDocument();
  });
  it("Language in component", () => {
    render(wrapper(<App />));
    screen.debug();
    expect(screen.getByText(/EN/i)).toBeInTheDocument();
  });
  it("Search has placeholder", () => {
    render(wrapper(<Search />));
    screen.debug();
    expect(
      screen.getByPlaceholderText(/Movies, person, movie, theaters/i)
    ).toBeInTheDocument();
  });
  it("Photocard return img with alt", () => {
    render(wrapper(<PhotoCard path={null} />));
    screen.debug();
    expect(screen.getByAltText(/PhotoCard/i)).toBeInTheDocument();
  });
  it("Metcblock view correct", () => {
    render(<MetaBlock meta={"meta text"} prefix={"prefix"} title={"title"} />);
    screen.debug();
    expect(screen.getByText(/meta text/i)).toBeInTheDocument();
    expect(screen.getByText(/prefix/i)).toBeInTheDocument();
    expect(screen.getByText(/title/i)).toBeInTheDocument();
  });
  it("Incorrect result path correct view", () => {
    render(wrapper(<IncorrectRequest path={"path"} />));
    screen.debug();
    expect(screen.getByText(/path/i)).toBeInTheDocument();
  });
  it("Language select view", () => {
    render(wrapper(<LanguageSelect language={"EN"} display={"ENGLISH"} />));
    screen.debug();
    expect(screen.getByText(/ENGLISH/i)).toBeInTheDocument();
  });
  it("People card is correct", () => {
    render(
      <People
        el={{ original_name: "Sara Connor", known_for_department: "HR" }}
      />
    );
    screen.debug();
    expect(screen.getByText(/Sara Connor/i)).toBeInTheDocument();
    expect(screen.getByText(/HR/i)).toBeInTheDocument();
  });
  it("Film card is correct", () => {
    const el = {
      title: "Title film",
      vote_average: 7.8,
    };
    render(wrapper(<FilmCard el={el} />));
    screen.debug();
    expect(screen.getByText(el.title)).toBeInTheDocument();
    expect(screen.getByText(el.vote_average.toFixed(1))).toBeInTheDocument();
    expect(screen.getByAltText(/Play/i)).toBeInTheDocument();
    expect(screen.getByAltText(el.title)).toBeInTheDocument();
  });
  it("Film cover is correct", () => {
    const el = {
      title: "Title film",
      vote_average: 7.8,
    };
    render(<FilmCover el={el} />);
    screen.debug();
    expect(screen.getByText(el.vote_average.toFixed(1))).toBeInTheDocument();
    expect(screen.getByAltText(/Cover/i)).toBeInTheDocument();
  });
  it("Tab is correct", () => {
    render(wrapper(<Tab display={["SOME", "SOME", "SOME"]} />));
    screen.debug();
    expect(screen.getByText(/some/i)).toBeInTheDocument();
  });
});

describe("Event testing", () => {
  it("Change pagination", () => {
    const handleClick = jest.fn();
    render(wrapper(<Paginations handler={handleClick} max={5} />));
    screen.debug();
    fireEvent.click(screen.getByText(/1/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
  it("Change pagination 2 level in", () => {
    const handleClick = jest.fn();
    render(wrapper(<Pagination handler={handleClick} num={"4"} />));
    screen.debug();
    fireEvent.click(screen.getByText(/4/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});

describe("Router testing", () => {
  it("No found page /hhhhh", () => {
    const { getByText } = render(routerWrapper("/hhhhh"));
    expect(getByText("Unfortunately, there is no page")).toBeInTheDocument();
  });
  it("No found page /movie", () => {
    const { getByText } = render(routerWrapper("/movie"));
    expect(getByText("Unfortunately, there is no page")).toBeInTheDocument();
  });
  it("Incorrect movie page /movie/451048r", () => {
    const { getByText } = render(routerWrapper("/movie/451048r"));
    expect(getByText("Invalid request")).toBeInTheDocument();
    expect(getByText("*/movie/")).toBeInTheDocument();
    expect(getByText("To main")).toBeInTheDocument();
  });
  it("Incorrect actor page /actor/5081trw", () => {
    const { getByText } = render(routerWrapper("/actor/5081trw"));
    expect(getByText("Invalid request")).toBeInTheDocument();
    expect(getByText("*/actor/")).toBeInTheDocument();
    expect(getByText("To main")).toBeInTheDocument();
  });
});
