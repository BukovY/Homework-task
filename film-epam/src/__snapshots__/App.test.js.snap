// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <App/> Component App view correct 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-bg-1 MuiPaper-elevation4"
  >
    <div
      className="MuiContainer-root makeStyles-flex-2 MuiContainer-maxWidthLg"
    >
      <p
        className="MuiTypography-root makeStyles-title-3 MuiTypography-body1"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Film App
        </a>
      </p>
      <form>
        <div
          className="MuiBox-root MuiBox-root-8 makeStyles-search-4"
        >
          <div
            className="MuiBox-root MuiBox-root-9 makeStyles-searchIcon-5"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            className="MuiInputBase-root makeStyles-inputRoot-6"
            onClick={[Function]}
          >
            <input
              aria-label="search"
              className="MuiInputBase-input makeStyles-inputInput-7"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Movies, person, movie, theaters"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
        >
          search
        </button>
      </form>
      <div
        className="MuiBox-root MuiBox-root-12"
      >
        <div
          className="MuiBox-root MuiBox-root-13 makeStyles-languageSelect-10"
          onClick={[Function]}
        >
          EN
        </div>
      </div>
    </div>
  </header>,
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      className="MuiBox-root MuiBox-root-16 makeStyles-root-15"
    >
      <div
        className="MuiButtonGroup-root MuiButtonGroup-contained"
        role="group"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Popular
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Top rated
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContained"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Upcoming
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-17"
    >
      <div
        className="MuiBox-root MuiBox-root-18 makeStyles-root-14"
      />
      <nav
        aria-label="pagination navigation"
        className="MuiPagination-root makeStyles-root-19"
      >
        <ul
          className="MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiPaginationItem-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-current="true"
              aria-label="page 1"
              className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined Mui-selected"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              1
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 2"
              className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              2
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 3"
              className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              3
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 4"
              className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              4
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 5"
              className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              5
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiPaginationItem-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>,
]
`;
